Build fails when deactivating visp_gui module

When deactivating the `visp_gui` module while compiling the `visp_rbt` module, the compilation of `visp_modules` fails with the following error:

```
/home/user/visp-ws/visp/modules/tracker/rbt/src/rendering/vpPanda3DDepthFilters.cpp:39:10: fatal error: visp3/gui/vpDisplayFactory.h: No such file or directory
   39 | #include <visp3/gui/vpDisplayFactory.h>
```

--- a/modules/tracker/rbt/src/rendering/vpPanda3DDepthFilters.cpp
+++ b/modules/tracker/rbt/src/rendering/vpPanda3DDepthFilters.cpp
@@ -36,8 +36,6 @@
 #include "graphicsEngine.h"
 #include "windowFramework.h"
 
-#include <visp3/gui/vpDisplayFactory.h>
-
 BEGIN_VISP_NAMESPACE
 #define VP_STRINGIFY(x) #x
 #define VP_TOSTRING(x) VP_STRINGIFY(x)

--- a/modules/tracker/rbt/test/catchRBTDataset.cpp
+++ b/modules/tracker/rbt/test/catchRBTDataset.cpp
@@ -259,12 +259,14 @@ SCENARIO("Running tracker on sequences with ground truth", "[rbt]")
       vpImage<unsigned char> displayI(h, w), displayDepth(h, w, 255), displayMask(h, w);
       vpImage<vpRGBa> displayRGB(h, w);
 
+#ifdef VISP_HAVE_MODULE_GUI
       std::vector<std::shared_ptr<vpDisplay>> displays = vpDisplayFactory::makeDisplayGrid(2, 2, 0, 0, 70, 70,
         "Gray", displayI,
         "Color", displayRGB,
         "Depth", displayDepth,
         "Mask", displayMask
       );
+#endif
 
       const std::vector<std::string> objectNames = { "dragon", "cube", "stomach", "lower_teeth" };
 

--- a/modules/tracker/rbt/test/test_utils.h
+++ b/modules/tracker/rbt/test/test_utils.h
@@ -5,7 +5,9 @@
 #include <visp3/ar/vpPanda3DGeometryRenderer.h>
 #include <visp3/ar/vpPanda3DRGBRenderer.h>
 
+#ifdef VISP_HAVE_MODULE_GUI
 #include <visp3/gui/vpDisplayFactory.h>
+#endif
 
 #include <vector>
 

--- a/modules/vision/test/keypoint-with-dataset/testKeyPoint-2.cpp
+++ b/modules/vision/test/keypoint-with-dataset/testKeyPoint-2.cpp
@@ -47,7 +47,9 @@
 
 #include <visp3/core/vpImage.h>
 #include <visp3/core/vpIoTools.h>
+#ifdef VISP_HAVE_MODULE_GUI
 #include <visp3/gui/vpDisplayFactory.h>
+#endif
 #include <visp3/io/vpImageIo.h>
 #include <visp3/io/vpParseArgv.h>
 #include <visp3/io/vpVideoReader.h>
@@ -174,7 +176,7 @@ void run_test(const std::string &env_ipath, bool opt_click_allowed, bool opt_dis
   vpDisplay *display = nullptr;
 
   if (opt_display) {
-#ifdef VISP_HAVE_DISPLAY
+#if defined(VISP_HAVE_DISPLAY) && defined(VISP_HAVE_MODULE_GUI)
     display = vpDisplayFactory::allocateDisplay(I, 0, 0, "ORB keypoints matching and pose estimation");
     display->setDownScalingFactor(vpDisplay::SCALE_AUTO);
 #else
@@ -324,7 +326,7 @@ void run_test(const std::string &env_ipath, bool opt_click_allowed, bool opt_dis
   keypoints.createImageMatching(I, IMatching);
 
   if (opt_display) {
-#ifdef VISP_HAVE_DISPLAY
+#if defined(VISP_HAVE_DISPLAY) && defined(VISP_HAVE_MODULE_GUI)
     display2 = vpDisplayFactory::allocateDisplay(IMatching, 0, static_cast<int>(I.getHeight()) / vpDisplay::getDownScalingFactor(I) + 30, "IMatching");
     display2->setDownScalingFactor(vpDisplay::SCALE_AUTO);
 #endif

--- a/modules/vision/test/keypoint-with-dataset/testKeyPoint-3.cpp
+++ b/modules/vision/test/keypoint-with-dataset/testKeyPoint-3.cpp
@@ -55,7 +55,9 @@
 
 #include <visp3/core/vpImage.h>
 #include <visp3/core/vpIoTools.h>
+#if defined(VISP_HAVE_MODULE_GUI)
 #include <visp3/gui/vpDisplayFactory.h>
+#endif
 #include <visp3/io/vpImageIo.h>
 #include <visp3/io/vpParseArgv.h>
 #include <visp3/io/vpVideoReader.h>
@@ -201,7 +203,7 @@ void run_test(const std::string &env_ipath, bool opt_click_allowed, bool opt_dis
   vpDisplay *display = nullptr;
 
   if (opt_display) {
-#ifdef VISP_HAVE_DISPLAY
+#if defined(VISP_HAVE_DISPLAY) && defined(VISP_HAVE_MODULE_GUI)
     display = vpDisplayFactory::allocateDisplay(Imatch, 0, 0, "ORB keypoints matching");
     display->setDownScalingFactor(vpDisplay::SCALE_AUTO);
 #else

--- a/modules/vision/test/keypoint-with-dataset/testKeyPoint-4.cpp
+++ b/modules/vision/test/keypoint-with-dataset/testKeyPoint-4.cpp
@@ -50,7 +50,9 @@
 #include <visp3/core/vpHomogeneousMatrix.h>
 #include <visp3/core/vpImage.h>
 #include <visp3/core/vpIoTools.h>
+#if defined(VISP_HAVE_MODULE_GUI)
 #include <visp3/gui/vpDisplayFactory.h>
+#endif
 #include <visp3/io/vpImageIo.h>
 #include <visp3/io/vpParseArgv.h>
 #include <visp3/io/vpVideoReader.h>
@@ -175,7 +177,7 @@ void run_test(const std::string &env_ipath, bool opt_click_allowed, bool opt_dis
     Imatch.resize(I.getHeight(), 2 * I.getWidth());
     Imatch.insert(I, vpImagePoint(0, 0));
 
-#ifdef VISP_HAVE_DISPLAY
+#if defined(VISP_HAVE_DISPLAY) && defined(VISP_HAVE_MODULE_GUI)
     display = vpDisplayFactory::allocateDisplay(I, 0, 0, "ORB keypoints matching");
     display->setDownScalingFactor(vpDisplay::SCALE_AUTO);
     display2 = vpDisplayFactory::allocateDisplay(Imatch, 0, static_cast<int>(I.getHeight()) / vpDisplay::getDownScalingFactor(I) + 40, "ORB keypoints matching");

--- a/modules/vision/test/keypoint-with-dataset/testKeyPoint-5.cpp
+++ b/modules/vision/test/keypoint-with-dataset/testKeyPoint-5.cpp
@@ -49,7 +49,9 @@
 
 #include <visp3/core/vpImage.h>
 #include <visp3/core/vpIoTools.h>
+#if defined(VISP_HAVE_MODULE_GUI)
 #include <visp3/gui/vpDisplayFactory.h>
+#endif
 #include <visp3/io/vpImageIo.h>
 #include <visp3/io/vpParseArgv.h>
 #include <visp3/vision/vpKeyPoint.h>
@@ -158,7 +160,7 @@ void run_test(const std::string &env_ipath, bool opt_click_allowed, bool opt_dis
   vpDisplay *display = nullptr;
 
   if (opt_display) {
-#ifdef VISP_HAVE_DISPLAY
+#if defined(VISP_HAVE_DISPLAY) && defined(VISP_HAVE_MODULE_GUI)
     display = vpDisplayFactory::allocateDisplay(I, 0, 0, "KeyPoints detection.");
 #else
     std::cout << "No image viewer is available..." << std::endl;

--- a/modules/vision/test/keypoint-with-dataset/testKeyPoint-6.cpp
+++ b/modules/vision/test/keypoint-with-dataset/testKeyPoint-6.cpp
@@ -47,7 +47,9 @@
 
 #include <visp3/core/vpImage.h>
 #include <visp3/core/vpIoTools.h>
+#if defined(VISP_HAVE_MODULE_GUI)
 #include <visp3/gui/vpDisplayFactory.h>
+#endif
 #include <visp3/io/vpImageIo.h>
 #include <visp3/io/vpParseArgv.h>
 #include <visp3/vision/vpKeyPoint.h>
@@ -204,7 +206,7 @@ void run_test(const std::string &env_ipath, bool opt_click_allowed, bool opt_dis
   vpDisplay *display = nullptr;
 
   if (opt_display) {
-#ifdef VISP_HAVE_DISPLAY
+#if defined(VISP_HAVE_DISPLAY) && defined(VISP_HAVE_MODULE_GUI)
     display = vpDisplayFactory::allocateDisplay(I, 0, 0, "KeyPoints detection.");
 #else
     std::cout << "No image viewer is available..." << std::endl;

--- a/modules/vision/test/keypoint-with-dataset/testKeyPoint.cpp
+++ b/modules/vision/test/keypoint-with-dataset/testKeyPoint.cpp
@@ -47,7 +47,9 @@
 
 #include <visp3/core/vpImage.h>
 #include <visp3/core/vpIoTools.h>
+#if defined(VISP_HAVE_MODULE_GUI)
 #include <visp3/gui/vpDisplayFactory.h>
+#endif
 #include <visp3/io/vpImageIo.h>
 #include <visp3/io/vpParseArgv.h>
 #include <visp3/io/vpVideoReader.h>
@@ -178,7 +180,7 @@ void run_test(const std::string &env_ipath, bool opt_click_allowed, bool opt_dis
   vpDisplay *display = nullptr;
 
   if (opt_display) {
-#ifdef VISP_HAVE_DISPLAY
+#if defined(VISP_HAVE_DISPLAY) && defined(VISP_HAVE_MODULE_GUI)
     display = vpDisplayFactory::allocateDisplay(Imatch, 0, 0, "ORB keypoints matching");
     display->setDownScalingFactor(vpDisplay::SCALE_AUTO);
 #else

